### PR TITLE
[BUG] Able to pass challenge "Basic JavaScript: Initializing Variables with the Assignment Operator" with incorrect answers #40406

### LINKED ISSUE #40406: [BUG] Able to pass challenge "Basic JavaScript: Initializing Variables with the Assignment Operator" with incorrect answers
**Describe your problem and how to reproduce it:**
The challenge asks the user to initialize the value a to 9. However, it seems like any value that starts with 9 will pass the test e.g. 90, 99, 9000, 9999999999.

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/initializing-variables-with-the-assignment-operator

**Tell us about your browser and operating system:**
* Browser Name: Firefox
* Browser Version: 83.0 (64-bit)
* Operating System: Windows 10 Home

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**
bug3
bug2
bug1

### PR DESCRIPTION
**Pull request by vbinithyanandamv**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #40406

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/initializing-variables-with-the-assignment-operator.md
Comment (Sky020) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/initializing-variables-with-the-assignment-operator.md:26]:
```suggestion
assert(/var\s+a\s*=\s*9(\s*;?\s*)$/.test(code));
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
This will still match `9.9999`

Also, please edit the body of this PR to include the related issue this PR is aimed at closing. For future reference, the issue had not been opened for a fix yet with the _help wanted_ or _first timers only_ labels - not much of a problem for this PR, but please wait for an issue to be triaged, before opening a PR for it.

If you have not already, please read the [Contributors Docs](https://contribute.freecodecamp.org/#/index). As it stands, this PR is going to soil your fork's `master` branch if it is merged. For future PRs, please follow the advice given in the docs about checking out a new branch to work off of.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/initializing-variables-with-the-assignment-operator.md
Comment (vbinithyanandamv) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/initializing-variables-with-the-assignment-operator.md:26]: Thanks for the suggestion @Sky020 . Will keep the triage label in mind next time. Also yea will make sure to create new branch for the fixes next time.

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍 

Please note: there is currently a Code Freeze on curriculum related PRs. So, only once that is lifted will this be merged. Also, any conflicts which could arise will need to be resolved.

**Comment by nhcarrigan** [comment]
I'd be okay with merging this as-is, to eliminate the bug.

Following that, I'd like to further discuss splitting this into two tests, so the value can be asserted directly and the `var` use can be checked with regex.

**Comment by vbinithyanandamv** [comment]
@nhcarrigan  thanks for the input. So this is going to be merged or going to put it up on hold?

**Comment by nhcarrigan** [comment]
At the moment we are in a code freeze and are not merging PRs to the curriculum challenges. 🙂

**Review by RandellDawson — APPROVED** [review]
LGTM

**Comment by RandellDawson** [comment]
Congrats @vbinithyanandamv on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Comment by vbinithyanandamv** [comment]
> Congrats @vbinithyanandamv on your first pull request (PR)! 🎉
> 
> Thank you for your contribution to the page! 👍
> We're happy to accept these changes, and look forward to future contributions. 📝

Thanks Randell and yea looking forward for more 👍

**Pull request merged**
